test_plans/multi_ptr_test_plan.asciidoc

```diff
@@ -0,0 +1,431 @@
+`multi_ptr<const value_type, access::address_space::global_space, IsDecorated>()`
+
+==== Explicit conversions to local_ptr
+
+For `access::address_space:generic_space` as Space parameter
+
+For the following values as DecorateAddress parameter:
+
+* `access::decorated::yes`
+* `access::decorated::no`
+
+For the following values as IsDecorated parameter:
+
+* `access::decorated::yes`
+* `access::decorated::no`
+
+Constract multi_ptr with accessor and check that it refers to the same value using get()
+and return type as raw_local_ptr or decorated_local_ptr for operators:
+
+* `multi_ptr<value_type, access::address_space::local_space, IsDecorated>()`
+* `multi_ptr<const value_type, access::address_space::local_space, IsDecorated>()`
+
+Check const correctness for operator
+`multi_ptr<const value_type, access::address_space::local_space, IsDecorated>()`
+
+
+==== Implicit conversions
+
+For the following values as Space parameter:
+
+* `access::address_space:global_space`
+* `access::address_space:local_space`
+* `access::address_space:private_space`
+* `access::address_space:generic_space`
+
+For the following values as DecorateAddress parameter:
+
+* `access::decorated::yes`
+* `access::decorated::no`
+
+For the following values as IsDecorated parameter:
+
+* `access::decorated::yes`
+* `access::decorated::no`
+
+Constract multi_ptr with accessor and check result implicit conversion to
+
+* `multi_ptr<const void, Space, IsDecorated>`
+* `multi_ptr<const value_type, Space, IsDecorated>`
+
+For `access::decorated::yes` as DecorateAddress parameter
+Constract multi_ptr with accessor and check result of implicit conversion to `multi_ptr<value_type, Space, access::decorated::no>`
+by checking that it refers to the same value using get()
+
+For `access::decorated::no` as DecorateAddress parameter
+Constract multi_ptr with accessor and check result of implicit conversion to `multi_ptr<value_type, Space, access::decorated::yes>`
+by checking that it refers to the same value using get()
+
+==== prefetch(size_t numElements)
+
+For `access::address_space:global_space` as Space parameter
+For the following values as DecorateAddress parameter:
+
+* `access::decorated::yes`
+* `access::decorated::no`
+
+Check call and const correctness prefetch(size_t numElements)
+
+==== Arithmetic operators
+
+For the following values as Space parameter:
+
+* `access::address_space:global_space`
+* `access::address_space:local_space`
+* `access::address_space:private_space`
+* `access::address_space:generic_space`
+
+For the following values as DecorateAddress parameter:
+
+* `access::decorated::yes`
+* `access::decorated::no`
+
+Constract multi_ptr with accessor to 5th element of array of 10 consecutive values starting with 1 of testing type
+with same template parameters and check result of operators:
+
+* `multi_ptr& operator++(multi_ptr& mp)` refers to `6`
+* `multi_ptr operator++(multi_ptr& mp)` refers to `6`
```

Comment:
```suggestion
* `multi_ptr operator++(multi_ptr& mp, int)` refers to `5`
```
https://en.cppreference.com/w/cpp/language/operator_incdec
But you need also to check in all these tests taking non-const references that the new value of `mp` and the value returned by the operation.